ui/src/app/groupings/[groupingPath]/layout.tsx

```diff
@@ -3,14 +3,24 @@ import ExportDropdown from './_components/export-dropdown';
 import GroupingHeader from './_components/grouping-header';
 import ReturnButtons from './_components/return-buttons';
 import SideNav from './_components/side-nav';
-import { groupingDescription } from '@/lib/fetchers';
+import { redirect } from 'next/navigation';
+import { groupingDescription, groupingPathIsValid, isAdmin, isGroupingOwner } from '@/lib/fetchers';
+import { getCurrentUser } from 'next-cas-client/app';
 
 const GroupingPathLayout = async ({ params, tab }: { params: { groupingPath: string }; tab: React.ReactNode }) => {
     const groupPath = decodeURIComponent(params.groupingPath);
     const { description } = await groupingDescription(groupPath);
     const groupName = groupPath.split(':').pop() as string;
     const fromManageSubject = groupPath.includes('manage-person');
 
+    const checkIsValidPath = await groupingPathIsValid(groupPath);
+    if (!checkIsValidPath) redirect('/');
+
+    const currentUser = await getCurrentUser();
+    const checkIsUserAdmin = await isAdmin(currentUser.uid);
+    const checkIsGroupingOwner = await isGroupingOwner(groupPath, currentUser.uid);
+    if (!checkIsUserAdmin && !checkIsGroupingOwner) redirect('/');
```

Comment:
You could condense this to:
```
if (!(await groupingPathIsValid(groupPath)) redirect('/');

const currentUser = await getCurrentUser();
if (!(await isAdmin(currentUser.uid)) && !(await isGroupingOwner(groupPath, currentUser.uid))) redirect('/');
```
Since you are not using the checkIsValidPath, checkIsUSerAdmin, or checkIsGroupingOwner variables elsewhere.